src/main/java/main/model/db/imports/HandlerFactory.java

```diff
@@ -4,24 +4,25 @@
 import main.model.db.imports.ImportHandlers.*;
 
 import java.io.File;
+import java.util.Arrays;
 import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
 
 class HandlerFactory {
+
     Handler getHandler(File file, String type, TestNameNodeType testNameNodeType, Date finishTime) throws AqualityException {
-        switch (ImportTypes.valueOf(type)){
+        validateTypeOnNameNodeRequirements(ImportTypes.valueOf(type), testNameNodeType);
+
+        switch (ImportTypes.valueOf(type)) {
             case MSTest:
-                if(testNameNodeType == null){
-                    throw new AqualityException("testNameNode is required");
-                }
                 return new TRX(file, testNameNodeType);
             case Robot:
                 return new Robot(file);
             case JUnit:
             case TestNG:
```

Comment:
could we remove Junit and TestNG from ImportTypes

Reply:
no, we keep these two types for backward campatibility